src/Infrastructure.EntityFramework/Repositories/GroupRepository.cs

```diff
@@ -84,6 +84,45 @@ public async Task<ICollection<Core.Entities.Group>> GetManyByOrganizationIdAsync
         }
     }
 
+    public async Task<ICollection<Tuple<Core.Entities.Group, ICollection<SelectionReadOnly>>>>
+        GetManyWithCollectionsByOrganizationIdAsync(Guid organizationId)
+    {
+        var groups = await GetManyByOrganizationIdAsync(organizationId);
+        using (var scope = ServiceScopeFactory.CreateScope())
+        {
+            var dbContext = GetDatabaseContext(scope);
+            var query = await (
+                from cg in dbContext.CollectionGroups
+                where cg.Group.OrganizationId == organizationId
+                select cg).ToListAsync();
+
+            return groups.Select(group =>
+                new Tuple<Core.Entities.Group, ICollection<SelectionReadOnly>>(
+                    group,
+                    query.Select(c => new SelectionReadOnly
+                    {
+                        Id = c.CollectionId,
+                        ReadOnly = c.ReadOnly,
+                        HidePasswords = c.HidePasswords,
+                    }).ToList()
+                )
+            ).ToList();
+        }
+    }
+
+    public async Task<ICollection<Core.Entities.Group>> GetManyByManyIds(IEnumerable<Guid> groupIds)
+    {
+        using (var scope = ServiceScopeFactory.CreateScope())
+        {
+            var dbContext = GetDatabaseContext(scope);
+            var query = from g in dbContext.Groups
+                        where groupIds.Contains(g.Id)
+                        select g;
+            var data = await query.ToArrayAsync();
```

Comment:
Why does this use `ToArray` instead of `ToList`, which is what we seem to use elsewhere here? (and is the more common choice afaik)

Reply:
I was just about to ask you to change this to a list, but you're way ahead of me ;)